app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -32,11 +32,16 @@ class BloodPressureEntrySheet : BottomSheetActivity() {
 
   companion object {
     private const val KEY_PATIENT_UUID = "patientUuid"
+    private const val EXTRA_WAS_BP_SAVED = "wasBpSaved"
 
-    fun intent(context: Context, patientUuid: UUID): Intent {
+    fun startForResult(activity: TheActivity, context: Context, patientUuid: UUID, requestCode: Int) {
       val intent = Intent(context, BloodPressureEntrySheet::class.java)
       intent.putExtra(KEY_PATIENT_UUID, patientUuid)
-      return intent
+      activity.startActivityForResult(intent, requestCode)
```

Comment:
Calling screen should be responsible for starting intents. The incoming screen shouldn't know about the `requestCode`. Can you move `startActivityForResult()` to summary screen?